src/Symfony/Component/Routing/Tests/Generator/Dumper/CompiledUrlGeneratorDumperTest.php

```diff
@@ -267,21 +272,18 @@ public function testAliases()
 
     public function testTargetAliasNotExisting()
     {
-        $this->expectException(RouteNotFoundException::class);
-
         $this->routeCollection->addAlias('a', 'not-existing');
 
         file_put_contents($this->testTmpFilepath, $this->generatorDumper->dump());
 
-        $compiledUrlGenerator = new CompiledUrlGenerator(require $this->testTmpFilepath, new RequestContext());
+        $this->expectException(RouteNotFoundException::class);
 
+        $compiledUrlGenerator = new CompiledUrlGenerator(require $this->testTmpFilepath, new RequestContext());
         $compiledUrlGenerator->generate('a');
     }
 
     public function testTargetAliasWithNamePrefixNotExisting()
```

Comment:
after investigating why testTargetAliasWithNamePrefixNotExisting are now failing I found out, that both, CompiledUrlGeneratorDUMPER and CompiledUrlGenerator throw a RouteNotFoundException. We are testing $compiledUrlGenerator->generate('sub_a) here, but the exception is already thrown in file_put_contents($this->testTmpFilepath, $this->generatorDumper->dump()); when generating the dump to prepare a file.

So now after moving the expectedException method call down, I can say, that we are actually not testing the behavior of CompiledUrlGenerator in this test 🤷‍♂️

Same applies to testTargetAliasNotExisting